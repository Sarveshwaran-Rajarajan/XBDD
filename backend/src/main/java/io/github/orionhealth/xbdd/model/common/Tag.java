package io.github.orionhealth.xbdd.model.common;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Tag {
	Integer line;
	String name;

	public Integer getLine() {
		return line;
	}

	public void setLine(final Integer line) {
		this.line = line;
	}

	public String getName() {
		return name;
	}

	public void setName(final String name) {
		this.name = name;
	}
}
