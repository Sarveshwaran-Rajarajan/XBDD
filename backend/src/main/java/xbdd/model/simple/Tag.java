package xbdd.model.simple;

public class Tag {
	Integer line;
	String name;

	public Integer getLine() {
		return line;
	}

	public void setLine(Integer line) {
		this.line = line;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}
}
