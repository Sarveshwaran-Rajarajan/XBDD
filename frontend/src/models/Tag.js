import SimpleFeature from "./SimpleFeature";

const getStatusPresences = features => {
  const findStatus = status => features.find(feature => feature.calculatedStatus === status);

  return {
    containsPassed: !!findStatus("passed"),
    containsUndefined: !!findStatus("undefined"),
    containsFailed: !!findStatus("failed"),
    containsSkipped: !!findStatus("skipped"),
  };
};

class Tag {
  constructor(data) {
    if (data) {
      this.name = data.tag;
      this.features = data.features.map(data => new SimpleFeature(data));
      Object.assign(this, getStatusPresences(data.features));
    }
  }

  clone() {
    const cloneTag = new Tag();
    cloneTag.name = this.name;
    cloneTag.features = this.features.map(simpleFeature => simpleFeature.clone());
    cloneTag.userName = this.userName;
    cloneTag.isIgnored = this.isIgnored;
    cloneTag.containsPassed = this.containsPassed;
    cloneTag.containsUndefined = this.containsUndefined;
    cloneTag.containsFailed = this.containsFailed;
    cloneTag.containsSkipped = this.containsSkipped;
    return cloneTag;
  }

  setUserForTag(userName) {
    this.userName = userName;
  }

  setUserForTags(data) {
    const tagAssignment = data.find(item => item.tag === this.name);
    if (tagAssignment) {
      this.userName = tagAssignment.userName;
    }
  }

  setIgnoredTags(data) {
    this.isIgnored = data.includes(this.name);
  }

  toggleIgnoreForTag() {
    this.isIgnored = !this.isIgnored;
  }
}

export default Tag;
