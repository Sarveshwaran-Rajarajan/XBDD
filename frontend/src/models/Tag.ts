import SimpleFeature from './SimpleFeature';

export type TagName = string;

export interface SimpleTag {
  name: TagName;
}

interface Tag extends SimpleTag {
  features: SimpleFeature[];
}

export default Tag;
