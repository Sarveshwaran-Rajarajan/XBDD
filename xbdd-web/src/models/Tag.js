const getStatusPresences = features => {
  const findStatus = status => features.find(feature => feature.calculatedStatus === status);

  return {
    containsPassed: !!findStatus("passed"),
    containsUndefined: !!findStatus("undefined"),
    containsFailed: !!findStatus("failed"),
    containsSkipped: !!findStatus("skipped"),
  };
};

class Tag {
  constructor(data) {
    this.name = data.tag;
    this.features = data.features;
    Object.assign(this, getStatusPresences(data.features));
  }
}

export default Tag;
