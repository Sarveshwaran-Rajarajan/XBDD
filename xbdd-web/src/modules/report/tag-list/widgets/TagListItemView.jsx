import React from "react";
import { PropTypes } from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import { ListItem, ListItemIcon, ListItemText, Collapse } from "@material-ui/core";
import { TurnedIn, ExpandLess, ExpandMore } from "@material-ui/icons";
import TagListFeaturesListView from "./TagListFeaturesListView";
import Tag from "../../../../models/Tag";
import { tagListItemStyles } from "../styles/TagListStyles";

const TagListItemView = props => {
  const { tag, isSelected, selectedStatus, handleTagSelect, classes } = props;
  const featureList = tag.features.filter(feature => selectedStatus[feature.calculatedStatus]);
  let className = classes.xbddTagListItemContainer;

  if (isSelected) {
    className += ` ${classes.xbddTagListItemContainerExpanded}`;
  }
  const onClick = () => handleTagSelect(tag.name);

  return (
    <>
      <ListItem button divider onClick={onClick} className={className}>
        <ListItemIcon className={classes.listItemIcon}>
          <TurnedIn />
        </ListItemIcon>
        <ListItemText>{tag.name}</ListItemText>
        {isSelected ? <ExpandLess /> : <ExpandMore />}
      </ListItem>
      <Collapse in={isSelected} timeout="auto" unmountOnExit>
        <TagListFeaturesListView featureList={featureList} />
      </Collapse>
    </>
  );
};

TagListItemView.propTypes = {
  tag: PropTypes.instanceOf(Tag).isRequired,
  isSelected: PropTypes.bool.isRequired,
  selectedStatus: PropTypes.shape({}).isRequired,
  handleTagSelect: PropTypes.func.isRequired,
  classes: PropTypes.shape({}).isRequired,
};

export default withStyles(tagListItemStyles)(TagListItemView);
