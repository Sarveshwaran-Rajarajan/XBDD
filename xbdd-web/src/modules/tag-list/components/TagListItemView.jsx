import React from 'react';
import { PropTypes } from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import ListItem from '@material-ui/core/ListItem';
import Tag from '../../../models/Tag';
import { tagListItemStyles } from '../styles/TagListStyles';

const TagListItemView = (props) => {
    const className = props.isSelected ? `${props.classes.xbddTagListItemContainer} ${props.classes.xbddTagListItemContainerSelected}` : props.classes.xbddTagListItemContainer;
    const onSelectTag = () => props.onSelectTag(props.tag);

    return (
        <ListItem
            button
            onClick={onSelectTag}
            className={className}
        >
            {props.tag.name}
        </ListItem>
    );
};

TagListItemView.propTypes = {
    tag: PropTypes.instanceOf(Tag).isRequired,
    classes: PropTypes.object.isRequired,
    onSelectTag: PropTypes.func.isRequired,
    isSelected: PropTypes.bool.isRequired,
};

export default withStyles(tagListItemStyles)(TagListItemView);
