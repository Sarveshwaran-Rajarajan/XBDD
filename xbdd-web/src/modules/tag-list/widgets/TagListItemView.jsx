import React from "react";
import { PropTypes } from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import ListItem from "@material-ui/core/ListItem";
import Tag from "../../../models/Tag";
import { tagListItemStyles } from "../styles/TagListStyles";

const TagListItemView = (props) => {
  const { tag, classes, onSelectTag, isSelected } = props;
  let className = classes.xbddTagListItemContainer;

  if (isSelected) {
    className += ` ${classes.xbddTagListItemContainerSelected}`;
  }
  const onClick = () => onSelectTag(props.tag);

  return (
    <ListItem button onClick={onClick} className={className}>
      {tag.name}
    </ListItem>
  );
};

TagListItemView.propTypes = {
  tag: PropTypes.instanceOf(Tag).isRequired,
  classes: PropTypes.shape({}).isRequired,
  onSelectTag: PropTypes.func.isRequired,
  isSelected: PropTypes.bool.isRequired,
};

export default withStyles(tagListItemStyles)(TagListItemView);
