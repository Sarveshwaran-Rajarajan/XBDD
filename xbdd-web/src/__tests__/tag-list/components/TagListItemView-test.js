import React from "react";
import sinon from "sinon";
import { shallow, mount } from "enzyme";
import TagListItemView from "../../../modules/tag-list/widgets/TagListItemView";
import Tag from "../../../models/Tag";

const dummyTag = new Tag({
    tag: "testTagOne",
    features: [],
});

describe("TagListItemView", () => {
    test("renders", () => {
        const t = true;
        const tagListItem = shallow(
            <TagListItemView tag={dummyTag} onSelectTag={() => {}} isSelected={t} />
        );
        expect(tagListItem).toMatchSnapshot();
    });

    test("onSelectTag function is called when tag is clicked", () => {
        const t = true;
        const onClick = sinon.spy();
        const tagListItem = mount(
            <TagListItemView tag={dummyTag} onSelectTag={onClick} isSelected={t} />
        );

        tagListItem.simulate("click");

        expect(onClick.called).toEqual(true);
    });
});
