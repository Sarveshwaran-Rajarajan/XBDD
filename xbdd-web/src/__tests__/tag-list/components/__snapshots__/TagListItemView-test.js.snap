// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagListItemView renders 1`] = `ShallowWrapper {}`;
